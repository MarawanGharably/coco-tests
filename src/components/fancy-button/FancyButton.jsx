import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/react';

const baseButtonCss = css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: inherit;
    height: inherit;
    background: #FF74A6;
    border: 1px solid #FF74A6;
    border-radius: 5px;
    &:hover {
        cursor: pointer;
    }
`;

const baseTextCss = css`
    font-size: 14px;
    font-weight: 600;
    color: #FFF;
    letter-spacing: 0.8px;
`;

// Fancy button uses emotion's style composition to receieve styles from props
const FancyButton = ({
    buttonStyle, textStyle, onClick, type, text,
}) => (
    // eslint-disable-next-line react/button-has-type
    <button css={[baseButtonCss, buttonStyle]} aria-label="fancy button" onClick={onClick} type={type}>
        <span css={[baseTextCss, textStyle]}>{text}</span>
    </button>
);

FancyButton.propTypes = {
    buttonStyle: PropTypes.func,
    onClick: PropTypes.func,
    type: PropTypes.string,
    textStyle: PropTypes.func,
    text: PropTypes.string,
};

FancyButton.defaultProps = {
    buttonStyle: null,
    onClick: null,
    type: null,
    textStyle: null,
    text: '',
};

export default FancyButton;
