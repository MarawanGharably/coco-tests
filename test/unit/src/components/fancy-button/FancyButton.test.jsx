import '@testing-library/jest-dom/extend-expect';
/** @jsx jsx */
import serializer from 'jest-emotion';
import { jsx, css } from '@emotion/react';
import { useState } from 'react';
import { act } from 'react-dom/test-utils';
import { render, fireEvent } from '@testing-library/react';
import FancyButton from '../../../../../src/components/fancy-button/FancyButton';

expect.addSnapshotSerializer(serializer);


const testText = 'Successful Click!';

const SpecialTestButton = () => {
    const [text, setText] = useState('text');

    const onClick = () => {
        setText(testText);
    };

    return (<FancyButton text={text} onClick={onClick} />);
};

describe('Fancy Button', () => {
    // Snapshot Unit test of Fancy Button
    test('Button Renders Correctly', () => {
        const button = render(<FancyButton buttonStyle={css`background-color: green;`} text="This is a button" />);
        expect(button).toMatchSnapshot();
    });
    // Unit test of onclick functionality, uses wrapper test component
    test('Text changes on click', () => {
        let component;

        act(() => {
            component = render(<SpecialTestButton />);
        });
        const { getByLabelText } = component;

        act(() => {
            fireEvent.click(getByLabelText('fancy button'));
        });

        expect(getByLabelText('fancy button')).toHaveTextContent(testText);
    });
});
