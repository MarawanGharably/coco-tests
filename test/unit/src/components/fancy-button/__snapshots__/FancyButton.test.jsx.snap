// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fancy Button Button Renders Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 8em;
  height: 3.7em;
  background: white;
  border: 1px solid black;
  background-color: green;
}

.emotion-0 {
  font-size: 14px;
  font-weight: 600;
}

<body>
    <div>
      <button
        aria-label="fancy button"
        class="emotion-1"
        type="button"
      >
        <span
          class="emotion-0"
        >
          This is a button
        </span>
      </button>
    </div>
  </body>,
  "container": .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 8em;
  height: 3.7em;
  background: white;
  border: 1px solid black;
  background-color: green;
}

.emotion-0 {
  font-size: 14px;
  font-weight: 600;
}

<div>
    <button
      aria-label="fancy button"
      class="emotion-1"
      type="button"
    >
      <span
        class="emotion-0"
      >
        This is a button
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
